src/app/digital/public/api/impl/DigitalComponent.ts

```diff
@@ -16,4 +18,34 @@ export class DigitalComponentImpl extends ComponentImpl<
     public override get info(): DigitalComponentInfo {
         return new DigitalComponentInfoImpl(this.circuit, this.kind);
     }
+
+    public override firstAvailable(portGroup: Port["group"]): DigitalPort | undefined {
+        const ports = this.internal.getPortsForComponent(this.id);
+
+        const inputTypes = this.info.inputPortGroups;
+        const outputTypes = this.info.outputPortGroups;
+
+        const isInputType = inputTypes.includes(portGroup);
+        const isOutputType = outputTypes.includes(portGroup)
+
+        if (!isInputType && !isOutputType) 
+            return undefined;
+
+        function firstAvailableHelper(comp: DigitalComponentImpl, id: string): Boolean {
```

Comment:
declare this as
```ts
const firstAvailableHelper = (id: string) => {
    ...
}
```
and then you can just use `this` and not pass in `comp` explicitly